flask_project/campaign_manager/models/models.py

```diff
@@ -348,6 +361,186 @@ def update(self, campaign_dto):
         self.create_on = datetime.now()
         session.commit()
 
+    def save_feature_types(self, data):
+        """
+            Creates or updates the campaign object in the database
+        """
+        # pre-processing featureTypes and feature-attributes
+        data_types = ast.literal_eval(data['types'])
+        for _type in data_types:
+            type_dict = data_types[_type]
+            name = type_dict['type']
+            feature = type_dict['feature']
+            featureType = FeatureType(
+                feature=feature,
+                name=name
+                )
+            featureType.create()
+            # commit attributes for the feature to the database
+            for _tag in type_dict['tags']:
+                attribute = Attribute(
+                    attribute_name=_tag
+                    )
+                attribute.create()
+                featureType.attributes.append(attribute)
+            session.commit()
+            self.feature_types.append(featureType)
+        session.commit()
+
+    def delete_feature_types(self):
+        feature_types = self.feature_types
+        for feature_type in feature_types:
+            feature_type.delete()
+        session.commit()
+
+    def save_geometry(self, data):
+        """
+            Creates or updates the campaign object in the database
```

Comment:
can you check all your comments. some of them dont describe at all what the function/method does.